### PR TITLE
Bug 1244232 - Add readme to functional tests folder

### PR DESCRIPTION
**Pull request by bobsilverberg**

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@mozilla/web-qa-sorcerers r?

This is pretty much a copy of the docs as they existed in the now dormant oneanddone-tests repo, with a couple of changes.

**Comment by rbillings** [comment]
Looks good. R+ :thumbsup:

File: oneanddone/tests/functional/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+Tests for oneanddone.mozilla.org
+================================
+
+This folder contains Selenium tests used to test the One and Done web site.
+
+[![license](https://img.shields.io/badge/license-MPL%202.0-blue.svg)](https://github.com/mozilla/oneanddone/blob/master/LICENSE)
+[![travis](https://img.shields.io/travis/mozilla/oneanddone-tests.svg?label=travis)](http://travis-ci.org/mozilla/oneanddone/)
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: https://img.shields.io/travis/mozilla/oneanddone.svg?label=travis

File: oneanddone/tests/functional/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Install the Python packages that are needed to run our tests using pip. In a terminal, 
+from the the project root, issue the following command:
+
+    pip install -Ur requirements/test.txt
+
+### Run the tests
+
+Tests are run using the command line. Below are a couple of examples of running the tests:
+
+To run all of the desktop tests against the development environment,
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: I don't think we need to call out 'desktop' as there are no mobile tests for O&D.

File: oneanddone/tests/functional/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+from the the project root, issue the following command:
+
+    pip install -Ur requirements/test.txt
+
+### Run the tests
+
+Tests are run using the command line. Below are a couple of examples of running the tests:
+
+To run all of the desktop tests against the development environment, 
+pass in the correct value for --baseurl, like so:
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: `--base-url`

File: oneanddone/tests/functional/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+	py.test --baseurl=https://oneanddone-dev.allizom.org --driver=firefox oneanddone/tests/functional/
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: Nit: We can use a bash code block, as we have done in other READMEs recently.

```bash
$ py.test --base-url https://oneanddone-dev.allizom.org --driver Firefox oneanddone/tests/functional/
```

Also note `--base-url` and `Firefox`, in recent READMEs we've also omitted the `=` for optional arguments.

File: oneanddone/tests/functional/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	
+You can also run the tests against a local instance of One and Done, if you have the project
+properly configured. To do so, the command line would be:
+
+	py.test  --driver=firefox oneanddone/tests/functional/
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: Nit: Extra space after `py.test`, and driver should be `Firefox`.

File: oneanddone/tests/functional/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+Note that in order to run the tests locally you also need to install the local test dependencies into
+your virtualenv, like so:
+
+    pip install -Ur requirements/test-local.txt
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: Nit: I feel like this should be before the command for running the tests locally.

File: oneanddone/tests/functional/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+The pytest plugin that we use for running tests has a number of advanced command-line 
+options available. To see the options available, try running:
+
+    py.test --help
+
+The full documentation for the plugin can be found at the [pytest-selenium GitHub project page] [pytestselenium].
+[pytestselenium]: https://github.com/davehunt/pytest-selenium
```
Comment (davehunt) [inline oneanddone/tests/functional/README.md]: In recent READMEs we've gathered the links at the bottom, which I find easier to maintain and makes the raw document easier to read. Just a suggestion though, no need to change if you prefer this way. See https://raw.githubusercontent.com/mozilla/sumo-tests/master/README.md for an example.

**Comment by bobsilverberg** [comment]
Thanks for the reviews @rbillings and @davehunt. I've pushed a new commit to address your suggestions.

**Comment by davehunt** [comment]
r+

**Pull request merged**